sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/BinaryOperationWithIdenticalExpressionsBase.cs

```diff
@@ -25,6 +25,8 @@ namespace SonarAnalyzer.Rules
     public abstract class BinaryOperationWithIdenticalExpressionsBase : SonarDiagnosticAnalyzer
     {
         internal const string DiagnosticId = "S1764";
-        internal const string MessageFormat = "Identical sub-expressions on both sides of operator '{0}'.";
+
+        protected const string OperatorMessageFormat = "Correct one of the identical expressions on both sides of operator '{0}'.";
+        protected const string EqualsMessage = "Correct one of the identical expressions on both sides of 'Object.Equals'.";

```

Comment:
this should probably be "... expressions in both arguments of ..." (if you change the message don't forget to update the unit tests and the integration test results)